packages/gatsby-plugin-mdx/gatsby-node.js

```diff
@@ -57,3 +57,58 @@ exports.onPostBootstrap = ({ cache }, pluginOptions) => {
     )
   }
 }
+
+if (process.env.GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION) {
+  exports.pluginOptionsSchema = function ({ Joi }) {
+    return Joi.object({
+      extensions: Joi.array()
+        .items(Joi.string())
+        .default([".mdx"])
+        .description(
+          `Configure the file extensions that gatsby-plugin-mdx will process`
+        ),
+      defaultLayout: Joi.object({})
+        .unknown(true)
+        .default({})
+        .description(`Set the layout components for MDX source types`),
+      gatsbyRemarkPlugins: Joi.array()
+        .items(
+          Joi.string(),
+          Joi.object({
+            resolve: Joi.string(),
+            options: Joi.object({}).unknown(true),
+          })
+        )
+        .default([])
+        .description(`Use Gatsby-specific remark plugins`),
+      remarkPlugins: Joi.array()
+        .items(
+          Joi.alternatives().try(
+            Joi.object({}).unknown(true),
+            Joi.array().items(Joi.object({}).unknown(true))
+          )
+        )
+        .default([])
+        .description(`Specify remark plugins`),
+      rehypePlugins: Joi.array()
+        .items(
+          Joi.alternatives().try(
+            Joi.object({}).unknown(true),
```

Comment:
We could eventually (not now!) think about introduciing a custom `Joi.rehype/remarkPlugin` type to further also verify this configuration 🤔

Reply:
Do you have some knowledge about these?